### PR TITLE
Input example

### PR DESCRIPTION
**Pull request by namnguyen20999**

### CODE REVIEW

File: doc/gui/examples/controls/input_active.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+init_value = 20
+
+page = """
+Input field has been disabled: <|{init_value}|input|active=false|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_active.py]:
```suggestion
<|{value}|input|active=false|>
```
Simpler is better.

File: doc/gui/examples/controls/input_active.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+init_value = 20
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_active.py]:
```suggestion
value = 20
```
Not 'init' for very long, normally...

File: doc/gui/examples/controls/input_active.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+page = """
+Input field has been disabled: <|{init_value}|input|active=false|>
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_active.py]:
```suggestion
    Gui(page).run(title="Input - Inactive")
```
That's new, from Fred. But I kind of like it.

File: doc/gui/examples/controls/input_change_delay_on_change.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+#     python <script>
+# -----------------------------------------------------------------------------------------
+import logging
+
+from taipy.gui import Gui
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+INIT_VALUE = ""
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_change_delay_on_change.py]:
```suggestion
value = ""
```
Capitals are somehow reserved for class members or constants.

File: doc/gui/examples/controls/input_change_delay_on_change.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+
+def on_change(state, var_name, value):
+    logging.info(f"Value of {var_name} changed to {value}")
+
+
+# If change_delay is set to -1, the change event is triggered on Enter key press
+page = """
+Enter a number (changes triggered on Enter key press): <|{INIT_VALUE}|input|change_delay=-1|on_change=on_change|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_change_delay_on_change.py]:
```suggestion
<|{value}|input|change_delay=-1|>
```

And add a little paragraph on the on_change setting:
"The [*on_change*][#p-on_change] property is not set: Taipy GUI automatically finds and binds the global *on_change()* function to handle changes in the input control. You could set this property to your own function if necessary."

File: doc/gui/examples/controls/input_change_delay_on_change.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+"""
+
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_change_delay_on_change.py]:
```suggestion
    Gui(page).run(title="input - on_change")
```

File: doc/gui/examples/controls/input_line_shown.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+numbered_lines = "1 \n2 \n3 \n4 \n5"
+
+page = """
+Multi-line input: <|{numbered_lines}|input|multiline|lines_shown=5|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_line_shown.py]:
```suggestion
<|{lines}|input|multiline|lines_shown=5|>
```

File: doc/gui/examples/controls/input_line_shown.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+numbered_lines = "1 \n2 \n3 \n4 \n5"
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_line_shown.py]:
```suggestion
lines = "1 \n2 \n3 \n4 \n5"
```

File: doc/gui/examples/controls/input_line_shown.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+page = """
+Multi-line input: <|{numbered_lines}|input|multiline|lines_shown=5|>
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_line_shown.py]:
```suggestion
    Gui(page).run(title="input - Lines shown")
```

File: doc/gui/examples/controls/input_multiline.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+LOREM_IPSUM = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam nec purus et nunc lacinia gravida. " \
+
+
+page = """
+Multi-line input: <|{LOREM_IPSUM}|input|multiline|>
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_multiline.py]:
```suggestion
    Gui(page).run(title="input - Multiline")
```

File: doc/gui/examples/controls/input_multiline.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+LOREM_IPSUM = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam nec purus et nunc lacinia gravida. " \
+
+
+page = """
+Multi-line input: <|{LOREM_IPSUM}|input|multiline|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_multiline.py]:
```suggestion
<|{text}|input|multiline|>
```

File: doc/gui/examples/controls/input_multiline.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+LOREM_IPSUM = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam nec purus et nunc lacinia gravida. " \
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_multiline.py]:
```suggestion
text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam nec purus et nunc lacinia gravida. " \
```

File: doc/gui/examples/controls/input_on_action.py [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+        logging.info("F1 key pressed")
+    elif key == 'F2':
+        logging.info("F2 key pressed")
+    elif key == 'F3':
+        logging.info("F3 key pressed")
+    else:
+        return None
+
+
+init_value = 0
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]:
```suggestion
value = 0
```

File: doc/gui/examples/controls/input_on_action.py [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+# on_action function is called when the action_keys are pressed
+page = """
+Enter a number (on_action triggered on F1, F2, F3 key press):
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]:
```suggestion
```

File: doc/gui/examples/controls/input_on_action.py [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+<|{init_value}|input|change_delay=300|on_action=print_value_on_key_pressed|action_keys=F1;F2;F3|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]:
```suggestion
<|{value}|input|change_delay=300|on_action=key_pressed|action_keys=F1;F2;F3|>
```

File: doc/gui/examples/controls/input_on_action.py [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+# -----------------------------------------------------------------------------------------
+import logging
+
+from taipy.gui import Gui
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def print_value_on_key_pressed(state, id, payload):
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]:
```suggestion
def key_pressed(state, id, payload):
```

File: doc/gui/examples/controls/input_password.py [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+page = """
+Input your password: <|input|password|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_password.py]:
```suggestion
<|input|password|>
```

File: doc/gui/examples/controls/input_password.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_password.py]:
```suggestion
    Gui(page).run(title="Input - Password")
```

File: doc/gui/examples/controls/input_on_action.py [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+init_value = 0
+
+# on_action function is called when the action_keys are pressed
+page = """
+Enter a number (on_action triggered on F1, F2, F3 key press):
+<|{init_value}|input|change_delay=300|on_action=print_value_on_key_pressed|action_keys=F1;F2;F3|>
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]:
```suggestion
    Gui(page).run(title="Input - On change")
```

File: doc/gui/examples/controls/input_type.py [hunk 20]
```diff
@@ -0,0 +1,25 @@
...
+from taipy.gui import Gui
+
+number = 0
+
+page = """
+Enter a number: <|{number}|input|type=number|>
+"""
+
+if __name__ == "__main__":
+    Gui(page).run()
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_type.py]:
```suggestion
    Gui(page).run(title="Input - Type")
```

File: doc/gui/examples/controls/input_type.py [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+number = 0
+
+page = """
+Enter a number: <|{number}|input|type=number|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_type.py]:
```suggestion
<|{number}|input|type=number|>
```

File: doc/gui/examples/controls/input_type.py [hunk 22]
```diff
@@ -0,0 +1,25 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+number = 0
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_type.py:18]: Isn't that some kind of reserved identifier?
What about 'value'?

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
To be more homogeneous - and more simple

File: doc/gui/examples/controls/input_multiline.py [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+text = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam nec purus et nunc lacinia gravida. " \
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_multiline.py]: "\\" at the end of the line?

File: doc/gui/examples/controls/input_on_action.py [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+
+from taipy.gui import Gui
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def key_pressed(state, id, payload):
+    key = payload.get('args', [None])[0]
+    if key == 'F1':
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/input_on_action.py]: Better?
```
if key in ["F1", "F2", "F3"]:
    logging.info(f"{key} key pressed")
```

Also, this function returns 'None'. Why?

**Review by FabienLelaquais — DISMISSED** [review]
A minor change. This looks great.

File: doc/gui/examples/controls/input_multiline.py (hunk 23 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/input_multiline.py]: removed

File: doc/gui/examples/controls/input_on_action.py (hunk 24 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/input_on_action.py]: resolved

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**